applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -404,46 +397,46 @@ namespace Kratos
 
 		KRATOS_TRY
 		const double numerical_limit = std::numeric_limits<double>::epsilon();
-		array_1d<double, msDimension> DirectionVectorX = ZeroVector(msDimension);
-		array_1d<double, msDimension> DirectionVectorY = ZeroVector(msDimension);
-		array_1d<double, msDimension> DirectionVectorZ = ZeroVector(msDimension);
-		array_1d<double, msLocalSize> ReferenceCoordinates = ZeroVector(msLocalSize);
-
-		ReferenceCoordinates[0] = this->GetGeometry()[0].X0();
-		ReferenceCoordinates[1] = this->GetGeometry()[0].Y0();
-		ReferenceCoordinates[2] = this->GetGeometry()[0].Z0();
-		ReferenceCoordinates[3] = this->GetGeometry()[1].X0();
-		ReferenceCoordinates[4] = this->GetGeometry()[1].Y0();
-		ReferenceCoordinates[5] = this->GetGeometry()[1].Z0();
+		array_1d<double, msDimension> direction_vector_x = ZeroVector(msDimension);

```

Comment:
i think there is an utility in the core which should centalize all of this. can u take a look?